src/robot-arm-element.ts

```diff
@@ -0,0 +1,106 @@
+import { html, LitElement } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { ElementPin } from '.';
+
+@customElement('wokwi-robot-arm')
+export class RobotArmElement extends LitElement {
+  /**
+   * Angle of shoulder
+   */
+  @property() sholderAngle = 297;
+
+  /**
+   * Angle of elbow
+   */
+  @property() elbowAngle = 52;
+
+  /**
+   * Angle of wrist
+   */
+  @property() wristAngle = 56;
+
+  /**
+   * Angle of gripper
+   */
+  @property() gripperAngle = 270;
+
+  /**
+   * The color of the gripper
+   */
+  @property() gripperColor = '#eeeedd';
+
+  /**
+   * The color of the arm
+   */
+  @property() armColor = '#ffffdd';
+
+  /**
+   * The color of the joints
+   */
+  @property() jointColor = '#111111';
+
+  readonly pinInfo: ElementPin[] = [
+    { name: 'PWM1', y: 195, x: 25, number: 1, signals: [{ type: 'pwm' }] },
+    { name: 'PWM2', y: 195, x: 30, number: 2, signals: [{ type: 'pwm' }] },
+    { name: 'PWM3', y: 195, x: 35, number: 3, signals: [{ type: 'pwm' }] },
+    { name: 'PWM4', y: 195, x: 40, number: 3, signals: [{ type: 'pwm' }] },
+  ];
+
+  render() {
+    /* eslint-disable prettier/prettier */
+    return html`
+      <style>
+        text {
+          user-select: none;
+        }
+      </style>
+      <svg version="1.1" width="70mm" height="52mm" viewBox="0 0 350 260" xml:space="preserve" xmlns="http://www.w3.org/2000/svg">
+          <!-- Terminals -->
+          <g>
+              <rect x="29" y="243" width="28" height="7" fill="${this.armColor}"/>
+              <rect x="32" y="250" width="1" height="9" fill="${this.jointColor}"/>
+              <rect x="38" y="250" width="1" height="9" fill="${this.jointColor}"/>
+              <rect x="45" y="250" width="1" height="9" fill="${this.jointColor}"/>
+              <rect x="52" y="250" width="1" height="9" fill="${this.jointColor}"/>
+          </g>
+          <g transform="translate(46.71 210.57)">
+              <!-- Shoulder -->
+              <g transform="rotate(${this.sholderAngle})">
+                  <path transform="translate(-1.27)" d="m1-36c-19.63 0-36.04 16.39-36.04 36s16.41 36 36.04 36 98.75-9 113.12-10c14.38-1 26.03-11 26.03-26s-11.65-24-26.03-26c-14.37-2-93.49-10-113.12-10z" fill="${this.armColor}" stroke="${this.jointColor}" stroke-miterlimit="10" stroke-width="2"/>
+
+                  <!-- Elbow -->
+                  <g transform="translate(113) rotate(${this.elbowAngle})">
```

Comment:
extracting the different values from `this` will probably make the code a bit tidies. i.e. at the start of the function:

```ts
const {elbowAngle, wristAngle, ...} = this;
```

Then you can use the variables directly, without repeating this. every time

Reply:
The more I modify SVG the harder it is to update the SVG if I want to add/remove stuff from it. When it become invalid SVG it is a lot harder to update